### PR TITLE
Global Function, Refresh Galleries and Albums, Code Cleanup and Other…

### PR DESCRIPTION
**Pull request by Code-BeautifulWeb**
@bradvin Please review the PR!

### CODE REVIEW

File: includes/objects/class-migratable.php [hunk 1]
```diff
@@ -78,6 +77,9 @@ function migrate_next_child() {
             if ( !$this->has_children() ) { return; }
             if ( $this->migration_status !== self::PROGRESS_ERROR && $this->migrated_child_count < $this->get_children_count() ) {
                 foreach ( $this->get_children() as $child ) {
+                    if($this->children_name() == 'galleries' && $this->get_total_images() == $this->get_total_migrated_images()) {
```
Comment (bradvin) [inline includes/objects/class-migratable.php:80]: Why is this check needed?

File: includes/plugins/class-photo.php [hunk 2]
```diff
@@ -199,29 +214,54 @@ private function get_photo_album( $id ) {
...
             if ( count( $photo_albums ) != 0 ) {
                 foreach ( $photo_albums as $key => $photo_album ) {
-                    $album = new Album( $this );
-                    $album->ID = $photo_album->id;
-                    $album->title = $photo_album->name;
-                    $album->data = $photo_album;
-                    $album->fooalbum_title = $photo_album->name;                        
+
+                    $unique_identifier = 'album_' . $this->name() . '_' . $photo_album->id;
+                    $migrated_object = foogallery_migrate_migrator_instance()->has_object_been_migrated( $unique_identifier );
```
Comment (bradvin) [inline includes/plugins/class-photo.php]: this code is redundant, as this logic now exists in get_album function

File: includes/objects/class-plugin.php [hunk 3]
```diff
@@ -72,5 +75,60 @@ function find_objects( $type ) {
...
+                $album->ID = $data['ID'];
+                $album->title = $data['title'];
+                $album->data = $data['data'];
+                $album->fooalbum_title = $data['fooalbum_title'];
+            }   
+
+            return $album;         
+        }        
+
+        function get_image( $data = array() ) {
```
Comment (bradvin) [inline includes/objects/class-plugin.php:131]: add PHP comments to the function

File: includes/objects/class-plugin.php [hunk 4]
```diff
@@ -72,5 +75,60 @@ function find_objects( $type ) {
...
+                $gallery->foogallery_title = $data['title'];
+                $gallery->data = $data['data'];
+                $gallery->children = $data['children'];
+                $gallery->settings = $data['settings'];
+            }   
+
+            return $gallery;         
+        }
+
+        function get_album( $data = array() ) {
```
Comment (bradvin) [inline includes/objects/class-plugin.php:109]: add PHP comments to the function

File: includes/objects/class-plugin.php [hunk 5]
```diff
@@ -72,5 +75,60 @@ function find_objects( $type ) {
             }
             return $this->find_galleries();
         }
+
+        function get_gallery( $data = array() ) {
```
Comment (bradvin) [inline includes/objects/class-plugin.php:84]: add php comments to function

**Review by bradvin — CHANGES_REQUESTED** [review]

File: includes/objects/class-migratable.php (hunk 1 above)
Comment (Code-BeautifulWeb) [reply includes/objects/class-migratable.php:80]: This is for the Album. If you migrate all Galleries and then migrate the album then the UI stops there so I added this condition.

File: includes/plugins/class-photo.php (hunk 2 above)
Comment (Code-BeautifulWeb) [reply includes/plugins/class-photo.php]: Sorry I forgot to remove this... Fixed in latest commit.

File: includes/objects/class-plugin.php (hunk 3 above)
Comment (Code-BeautifulWeb) [reply includes/objects/class-plugin.php:131]: Updated in latest commit

File: includes/objects/class-plugin.php (hunk 4 above)
Comment (Code-BeautifulWeb) [reply includes/objects/class-plugin.php:109]: Updated in latest commit

File: includes/objects/class-plugin.php (hunk 5 above)
Comment (Code-BeautifulWeb) [reply includes/objects/class-plugin.php:84]: Updated in latest commit

**Pull request merged**